File: LocalisationMateriel/src/java/metier/TypeServiceImpl.java
package metier;

import java.util.ArrayList;
import java.util.List;
import physique.PhysiqueFactory;
import physique.TypeDataService;

public class TypeServiceImpl implements TypeService {

    private final TypeDataService typeDataSrv;
    private final MaterielService materielSrv;

    TypeServiceImpl() {
        typeDataSrv = PhysiqueFactory.getTypeDataService();
        materielSrv = MetierFactory.getMaterielService();
    }

    @Override
    public Type add(Type type) {
        Type typ = null;
        try {
            if (!type.getNomType().isEmpty() && !type.getNomImage().isEmpty()) {
                typ = typeDataSrv.add(type);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return typ;
    }

    @Override
    public boolean remove(Type type) {
        boolean execute = false;
        try {
            if (type.getId() > 0) {
                List<Materiel> materiels = materielSrv.getByType(type, 0, materielSrv.getCountByType(type));
                if(materiels.isEmpty()){
                    execute = typeDataSrv.remove(type);
                }
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return execute;
    }

    @Override
    public boolean update(Type type) {
        boolean execute = false;
        try {
            if (type.getId() > 0) {
                if (!type.getNomType().isEmpty() && !type.getNomImage().isEmpty()) {
                    execute = typeDataSrv.update(type);
                }
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return execute;
    }

    @Override
    public List<Type> getAll(int debut, int parPage) {
        List<Type> typ = new ArrayList<Type>();
        try {
            if (debut >= 0 && parPage > 0) {
                typ = typeDataSrv.getAll(debut, parPage);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return typ;
    }

    @Override
    public int getCount() {
        int count = 0;
        try {
            count = typeDataSrv.getCount();
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return count;
    }

    @Override
    public Type getById(long id) {
        Type typ = null;
        try {
            if (id > 0) {
                typ = typeDataSrv.getById(id);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return typ;
    }

    @Override
    public Type getByNomType(String nomType) {
        Type typ = null;
        try {
            if (!nomType.isEmpty()) {
                typ = typeDataSrv.getByNomType(nomType);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return typ;
    }

    @Override
    public List<String> getListNomType(int debut, int parPage) {
        List<String> listStr = new ArrayList<String>();
        try {
            if (debut >= 0 && parPage > 0) {
                listStr = typeDataSrv.getListNomType(debut, parPage);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return listStr;
    }

    @Override
    public int getCountListNomType() {
        int count = 0;
        try {
            count = typeDataSrv.getCountListNomType();
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return count;
    }

    @Override
    public Type getByNomImage(String nomImage) {
        Type typ = null;
        try {
            if (!nomImage.isEmpty()) {
                typ = typeDataSrv.getByNomImage(nomImage);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return typ;
    }

    @Override
    public List<String> getListNomImage(int debut, int parPage) {
        List<String> listStr = new ArrayList<String>();
        try {
            if (debut >= 0 && parPage > 0) {
                listStr = typeDataSrv.getListNomImage(debut, parPage);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return listStr;
    }

    @Override
    public int getCountListNomImage() {
        int count = 0;
        try {
            count = typeDataSrv.getCountListNomImage();
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return count;
    }
}

File: AppMateriel/metier/typeserviceimpl.cpp
#include "typeserviceimpl.h"

Metier::TypeServiceImpl::TypeServiceImpl()
{
    typeDataSrv = PhysiqueFactory::getTypeDataServiceImpl();
    materielSrv = MetierFactory::getMaterielServiceImpl();
}

Type Metier::TypeServiceImpl::add(Type type)
{
    Type typ;
    if(!type.getNomType().isEmpty() && !type.getNomImage().isEmpty()){
        typ = typeDataSrv->add(type);
    }
    return typ;
}

bool Metier::TypeServiceImpl::remove(Type type)
{
    bool execute = false;
    if(type.getId() > 0){
        QList<Materiel> materiels = materielSrv->getByType(type, 0, materielSrv->getCountByType(type));
        if(materiels.isEmpty()){
            execute = typeDataSrv->remove(type);
        }
    }
    return execute;
}

bool Metier::TypeServiceImpl::update(Type type)
{
    bool execute = false;
    if(type.getId() > 0){
        if(!type.getNomType().isEmpty() && !type.getNomImage().isEmpty()){
            execute = typeDataSrv->update(type);
        }
    }
    return execute;
}

QList<Type> Metier::TypeServiceImpl::getAll(int debut, int parPage)
{
    QList<Type> typ;
    if(debut >= 0 && parPage > 0){
        typ = typeDataSrv->getAll(debut, parPage);
    }
    return typ;
}

int Metier::TypeServiceImpl::getCount()
{
    return typeDataSrv->getCount();
}

Type Metier::TypeServiceImpl::getById(long id)
{
    Type typ;
    if(id > 0){
        typ = typeDataSrv->getById(id);
    }
    return typ;
}

Type Metier::TypeServiceImpl::getByNomType(QString nomType)
{
    Type typ;
    if(!nomType.isEmpty()){
        typ = typeDataSrv->getByNomType(nomType);
    }
    return typ;
}

QList<QString> Metier::TypeServiceImpl::getListNomType(int debut, int parPage)
{
    QList<QString> listStr;
    if(debut >= 0 && parPage > 0){
        listStr = typeDataSrv->getListNomType(debut, parPage);
    }
    return listStr;
}

int Metier::TypeServiceImpl::getCountListNomType()
{
    int count = typeDataSrv->getCountListNomType();
    return count;
}

Type Metier::TypeServiceImpl::getByNomImage(QString nomImage)
{
    Type typ;
    if(!nomImage.isEmpty()){
        typ = typeDataSrv->getByNomImage(nomImage);
    }
    return typ;
}

QList<QString> Metier::TypeServiceImpl::getListNomImage(int debut, int parPage)
{
    QList<QString> listStr;
    if(debut >= 0 && parPage > 0){
        listStr = typeDataSrv->getListNomImage(debut, parPage);
    }
    return listStr;
}

int Metier::TypeServiceImpl::getCountListNomImage()
{
    int count = typeDataSrv->getCountListNomImage();
    return count;
}
